packages/hooks/use-post-type-infinite-scroll/types.ts

```diff
@@ -0,0 +1,97 @@
+import { IntersectionOptions } from "../use-infinite-scroll/types";
+
+/**
+ * Options received by the {@link usePostTypeInfiniteScroll} hook.
+ */
+export type UsePostTypeInfiniteScrollOptions = {
+  /**
+   * The number of pages that the hook should load automatically before
+   * switching to manual fetching.
+   */
+  limit?: number;
+
+  /**
+   * A boolean indicating if this hook should be active or not. It can be
+   * useful in situations where users want to share the same component for
+   * different types of Archives, but avoid doing infinite scroll in some of
+   * them.
+   */
+  active?: boolean;
+
+  /**
+   * The archive that should be used to get the next posts. If none is present,
+   * the previous link is used. If the previous link is not an archive, the
+   * homepage is used.
+   */
+  archive?: string;
+
+  /**
+   * The archive that should be used if the `archive` option is not present and
+   * the previous link is not an archive.
+   */
+  fallback?: string;
+
+  /**
+   * The intersection observer options for fetching.
+   */
+  fetchInViewOptions?: IntersectionOptions;
+
+  /**
+   * The intersection observer options for routing.
+   */
+  routeInViewOptions?: IntersectionOptions;
+};
+
+/**
+ * Object returned by the {@link usePostTypeInfiniteScroll} hook.
+ */
+export type UsePostTypeInfiniteScrollOutput = {
+  /**
+   * An array of the existing posts. Users should iterate over this array in
+   * their own layout.
+   */
+  posts: {
+    /**
+     * A unique key to be used in the iteration.
+     */
+    key: string;
+
+    /**
+     * The link of this post.
+     */
+    link: string;
+
+    /**
+     * If this post is the last post. Useful to add separators between posts,
+     * but avoid adding it for the last one.
+     */
+    isLast: boolean;
+
+    /**
+     * The Wrapper component that should wrap the real `Post` component.
+     */
+    Wrapper: React.FC;
```

Comment:
Can we improve the type of the final `Wrapper` components?

I was thinking about at least a `key` prop, but I want to know your opinion: 

```ts
Wrapper: React.FC<{ key: string; }>;
```

Reply:
Yes, that would be a good idea. We could use that `Wrapper` type here and also move these props to it.

https://github.com/frontity/frontity/blob/09ec9433b5b74fc6435a30326697971d9a36febc/packages/hooks/use-post-type-infinite-scroll/index.tsx#L31-L36

By the way, I see what you did. You refactored the way `Wrapper` components are generated and moved that logic to some utils inside the `use-infinite-scroll` folder. :raised_hands: 

My only concern is that it could be confusing in some way because the exported `Wrapper` component is in fact a kind of internal wrapper used inside the final `Wrapper` that developers get when calling `useArchiveInfiniteScroll` or `usePostTypeInfiniteScroll`.

So, instead of using `Wrapper` component and `WrapperProps` for those internal things, would it make sense if we reserve those names for the `Wrapper` components returned by the hooks?

Reply:
Absolutely. Good point.

Can you make the rename?

Reply:
Yup, I'll try to find a good name. :+1:

Reply:
I renamed it to `InternalWrapper` and `InternalWrapperProps`, and created `WrapperProps`, which is the type used by the `<Wrapper>` components returned by the hooks.

Reply:
Awesome. Thanks a lot David!